packages/hierarchies/src/hierarchies/internal/operators/EachValueFrom.ts

```diff
@@ -0,0 +1,128 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Observable } from "rxjs";
+
+/**
+ * This is pretty much a copy of https://github.com/benlesh/rxjs-for-await/blob/94f9cf9cb015ac3700dfd1850eb81d36962eb70f/src/index.ts#L33,
+ * except that we're using a linked list-based queue rather than an array-based one.
+ *
+ * @internal
+ */
+export async function* eachValueFrom<T>(source: Observable<T>): AsyncIterableIterator<T> {
+  const deferreds = new Queue<ResolvablePromise<{ value?: T; done: boolean }>>();
```

Comment:
Is there a point for copying the exact same code? 
`deferred` variable can be renamed to something like `deferredPromises`, `null` can be changed to `undefined`, and all `else`s in `while(true)` can be removed.

Reply:
don't see a point in renaming `deferred` to `deferredPromise`, but agree about the rest - cleaned up

Reply:
Sorry, I meant `deferreds` variable which is not a word and was added to the spell checker exceptions

Reply:
I think that's valid word: https://www.dictionary.com/browse/deferred

Reply:
It says that deferred is an adjective and here it's used as a noun in plural. I'm not sure if it's valid

Reply:
let's try this, then :) https://en.wiktionary.org/wiki/deferreds#English